lib/IDE/IDETypeChecking.cpp

```diff
@@ -667,6 +667,10 @@ class ExpressionTypeCollector: public SourceEntityWalker {
     if (E->getType().isNull())
       return false;
 
+    // We should not report a type for InjectIntoOptionalExpr
+    if (isa<InjectIntoOptionalExpr>(E))
+      return false;
```

Comment:
[Cursor-info ignores all `isImplicit()` expressions](https://github.com/apple/swift/blob/c50263c7301dd47b2982041d44ca8eb46504e7cb/lib/IDE/IDERequests.cpp#L301-L302). Could you try:
```suggestion
    if (E->isImplict())
      return false;
```

Reply:
Yes, that works, but I think we should add an exception for `OptionalEvaluationExpr`, right?

Reply:
Also, currently, in case of an implicit `DotSyntaxCallExpr` like in `let a = 1 + 2`, we expect the type of `+` to be `(Int, Int) -> Int`. With your suggested change that becomes `(Int.Type) -> (Int, Int) -> Int`. Is that correct, or do you think we should also opt in for reporting types of implicit `DotSyntaxCallExpr`?